cmd/upgrade.go

```diff
@@ -199,6 +197,7 @@ func (uc *upgradeCmd) validateCurrentLocalState(armTemplateHandle io.Reader) err
 			return errors.Wrap(err, "Invalid upgrade target version. Consider using --force if you really want to proceed")
 		}
 	}
+	uc.containerService.Properties.OrchestratorProfile.OrchestratorVersion = uc.upgradeVersion
```

Comment:
Doing this in the validation function `validateCurrentLocalState` is still ambiguous, can we do it when the function returns?

Reply:
otherwise maybe rename the function to reflect that it does more than validation

Reply:
I tried, but realized it wasn't achieving much, because lots of other areas are setting state in there.
it also made testing much harder, I had a test harness for the validate function, but there isn't any for loadCluster I think, which will need more setup.
I'll change the name of the method, getting a test harness around the loadCluster function is out of scope for this fix.

Reply:
Maybe `intializeFromLocalState` is more accurate ?
It sets the namesuffix, it sets the agentPools name, it sets the targetVersion.
Validation is secondary when you start setting state all over the place :)